auth/simple_token.go

```diff
@@ -21,13 +21,104 @@ import (
 	"crypto/rand"
 	"math/big"
 	"strings"
+	"sync"
+	"time"
 )
 
 const (
 	letters                  = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
 	defaultSimpleTokenLength = 16
+	simpleTokenTTL           = 5 * time.Minute
+	simpleTokenTTLResolution = 1 * time.Second
 )
 
+type simpleTokenTTLKeeper struct {
+	tokens              map[string]time.Time
+	addSimpleTokenCh    chan string
+	resetSimpleTokenCh  chan string
+	deleteSimpleTokenCh chan string
+	stopCh              chan chan struct{}
+	deleteTokenFunc     func(string)
+	mu                  sync.Mutex
+	running             bool
```

Comment:
Probably it is useful for stopping a goroutine of `simpleTokenTTLKeeper.run()`? I think `stopCh` is needless if we can change the line https://github.com/coreos/etcd/pull/6574/files#diff-dd4312fcf7448981a67eabc41a2019f3R93 to like this: `for running {`.